### PR TITLE
[sdk#898] Add clockmock Start

### PR DESCRIPTION
**Pull request by Bolodya1997**
## Description
1. Add clockmock Start (needed for the discover stage in sandbox tests).
2. Fix clocmock race issues.

## Issue link
Relates to #898.

## How Has This Been Tested?

- [x] Added unit testing to cover
- [x] Tested manually
- [ ] Tested by integration testing
- [ ] Have not tested

## Types of changes

- [x] Bug fix
- [x] New functionallity
- [ ] Documentation
- [ ] Refactoring
- [ ] CI

### CODE REVIEW

File: pkg/tools/clockmock/mock.go [hunk 1]
```diff
@@ -42,20 +42,48 @@ func New() *Mock {
 	}
 }
 
+// Start starts mock time to run with the given speed until ctx becomes done. While time is running, current time for
+// the mock will be the following:
+//   mock time := mock start time  +  (real time duration from the start) * speed  +  mock duration added with Set, Add
+func (m *Mock) Start(ctx context.Context, speed float64) {
```
Comment (denis-tingaikin) [inline pkg/tools/clockmock/mock.go:48]: 1. Do we need stop method?
2. What if Start has called twice?

Comment (Bolodya1997) [reply pkg/tools/clockmock/mock.go:48]: 1. Nope, because we have ctx.
2. Time will run with (speed-1 + speed-2) speed :)

Comment (denis-tingaikin) [reply pkg/tools/clockmock/mock.go:48]: Should this operation be thread-blockable?

File: pkg/tools/clockmock/mock.go [hunk 2]
```diff
@@ -42,20 +42,48 @@ func New() *Mock {
...
+	m.lock.RLock()
+	defer m.lock.RUnlock()
+
+	const tick = 10 * time.Millisecond
```
Comment (denis-tingaikin) [inline pkg/tools/clockmock/mock.go:52]: Should tick be configurable?

Comment (Bolodya1997) [reply pkg/tools/clockmock/mock.go:52]: It can, but I can get no usecase why it should.

File: pkg/tools/clockmock/mock.go (hunk 1 above)
Comment (Bolodya1997) [reply pkg/tools/clockmock/mock.go]: Reworked with always running time and `Mock.SetSpeed`.

**Review by d-uzlov — APPROVED** [review]

**Review by denis-tingaikin — APPROVED** [review]

**Pull request merged**